src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs

```diff
@@ -175,16 +179,17 @@ private Task ProcessNegotiationRequest(IOwinContext owinContext, HostContext con
 
             try
             {
-                accessToken = provider.GenerateClientAccessToken(null, claims);
+                accessToken = await provider.GenerateClientAccessToken(null, claims);
```

Comment:
nit: extract to another method so that you don't need to change others to `await`

Reply:
We can't.
Every place where is currently using `AccessToken` should wait for its authorization.